packages/@aws-cdk/aws-cfn-include/README.md

```diff
@@ -0,0 +1,190 @@
+with existing CloudFormation templates in the CDK.
+It can be thought of as an extension of the capabilities of the
+[`CfnInclude` class](../@aws-cdk/core/lib/cfn-include.ts).
+
+## Basic usage
+
+Let's say you have the file `my-template.json` with the following contents:
+
+```json
+{
+  "Resources": {
+    "Bucket": {
+      "Type": "AWS::S3::Bucket",
+      "Properties": {
+        "BucketName": "some-bucket-name"
+      }
+    }
+  }
+}
+```
+
+You can include it in your CDK application using the following code:
+
+```typescript
+import * as inc from '@aws-cdk/aws-cfn-include';
+
+const cfnTemplate = inc.CdkInclude.includeJsonTemplate(this, 'my-template.json');
+```
+
+That will add all resources from `my-template.json`
+into your CDK application,
+as children of `this`.
+
+You can retrieve any resource from the included template by referring to
+it by its logical ID.
+If you know the class of the CDK object that corresponds to that resource,
+you can cast the returned object to the correct type:
+
+```typescript
+import * as s3 from '@aws-cdk/aws-s3';
+
+const cfnBucket = cfnTemplate.getResource('Bucket') as s3.CfnBucket;
+// cfnBucket is of type s3.CfnBucket
+```
+
+Any modifications you make to that resource will be reflected in the resulting CDK template;
+for example, you can change the name of the bucket:
+
+```typescript
+cfnBucket.bucketName = 'my-bucket-name';
+```
+
+You can also refer to the resource when defining other resources,
+for example:
+
+```typescript
+import * as iam from '@aws-cdk/aws-iam';
+
+const role = new iam.Role(this, 'Role', {
+  assumedBy: new iam.AnyPrincipal(),
+});
+role.addToPolicy(new iam.PolicyStatement({
+  actions: ['s3:*'],
+  resources: [cfnBucket.attrArn],
+}));
+```
+
+You can also import a higher-level construct,
+by referring to the properties of the lower-level resource from the template:
+
+```typescript
+const bucket = s3.Bucket.fromBucketName(this, 'L2Bucket', cfnBucket.ref);
+// bucket is of type s3.IBucket
+```
+
+## Merging templates
+
+Imagine that, in addition to the `my-template.json` file above,
+you also had a different CloudFormation template file,
+`other-template.json`, with the contents:
+
+```json
+{
+  "Resources": {
+    "Bucket": {
+      "Type": "AWS::S3::Bucket",
+      "Properties": {
+        "BucketName": "other-bucket-name"
+      }
+    }
+  }
+}
+```
+
+Normally, you couldn't use both of these templates in the same CloudFormation stack,
+as they both define a resource with the same logical ID, `"Bucket"`.
+However, using CDK's construct hierarchy,
+it's possible to include both templates in your CDK application -
+you just have to make sure they're imported under different scopes!
+
+```typescript
+import { Construct } from '@aws-cdk/core';
+
+const scope1 = new Construct(this, 'Scope1');
+const scope2 = new Construct(this, 'Scope2');
+
+const cfnTemplate1 = inc.CdkInclude.includeJsonTemplate(scope1, 'my-template.json');
```

Comment:
Reiterating my comment from above: if `CdkInclude` would have just been a normal construct, this would have been natural.

Reply:
Done.